readthedocs/doc_builder/environments.py

```diff
@@ -728,6 +728,10 @@ def create_container(self):
                         'bind': self.project.doc_path,
                         'mode': 'rw'
                     },
+                    self.project.pip_cache_path: {
```

Comment:
> This is slightly redundant, as the project pip cache path is under doc_path already.

Mmm... Yes. This is redundant when the `GLOBAL_PIP_CACHE` is not set. So, maybe we should add this path only when it's set (as you said, conditionally).

> Also, normally to share a pip cache, we'd set CACHE_DIR env variable, or --cache-dir argument on pip

We are using the `--cache-dir` in the command at `python_environments.py` file, and that dir points to the `project.pip_cache_path`

I prefer to use the argument option instead of the env variable (since it's _less_ explicit and more complicated to debug later)

> I don't know enough about pip internals to answer this

As far as I know, pip doesn't save metadata, but creates a hash of the package downloaded and use it as filename --but use the real one for wheels:

```
├── http
│   ├── 0
│   │   ├── 0
│   │   │   ├── 3
│   │   │   │   └── f
│   │   │   │       └── 8
│   │   │   │           └── 003f82596103cb4b5d71b69aeb8119f406b1bc78110a190848c4fb24
│   │   │   ├── 6
│   │   │   │   └── 6
│   │   │   │       └── b
│   │   │   │           └── 0066b726b642ce2657f32a7c7b9f63ae698a2a16695cda32f18eb806
```

```
└── wheels
    ├── 02
    │   └── 2e
    │       └── 0d
    │           └── 30ef9919ce3a07990f191c164d07b083761535fb721a1af3cf
    │               ├── ip_associations_python_novaclient_ext-0.2-cp27-none-any.whl
    │               └── ip_associations_python_novaclient_ext-0.2-cp36-none-any.whl
    ├── 06
    │   └── 3d
    │       └── 74
    │           └── f64c3ce3266df2598031818ecc86bc25c8ccede0723973f119
    │               └── anyjson-0.3.3-cp27-none-any.whl
```

I'd say that mounting conditionally under the binds depending on this setting is the missing piece here and the rest is safe.

Reply:
This sounds good. I don't *think* there is any metadata dependent on the path either, so this is most likely safe I'm sure.